### PR TITLE
feat: JunoPay onramp

### PR DESCRIPTION
**Pull request by rakeshnuo**
## Description

Adds JunoPay onramp to fiat onramps.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Testing
To test, use the JunoPay fiat onramp and it should direct you to buy crypto through your Juno account. The crypto should be sent to your ShapeShift wallet.

## Screenshots (if applicable)
Screenshot 2022-07-07 at 12 17 50 AM
Screenshot 2022-07-07 at 12 18 26 AM
Screenshot 2022-07-07 at 8 18 41 PM

### CODE REVIEW

**Comment by gomesalexandre** [comment]
This is currently blocked by https://github.com/shapeshift/lib/pull/863/ - moving it to draft until it's published as CI won't pass

File: .github/workflows/pr-cypress.yml [hunk 1]
```diff
@@ -30,6 +30,11 @@ env:
   REACT_APP_COINBASE_PAY_APP_ID: 1dbd2a0b94
   REACT_APP_GEM_ENV: production
 
+  REACT_APP_JUNOPAY_APP_ID: live_SYkQkrlyIQeuQf8AauSOaGTa
+  REACT_APP_JUNOPAY_ASSET_LOGO: https://cdn.juno.finance/
+  REACT_APP_JUNOPAY_BASE_URL: https://api.juno.finance
+  REACT_APP_JUNOPAY_SUPPORTED_COINS: https://api.juno.finance/crypto-wallet-partners?partner_key=live_SYkQkrlyIQeuQf8AauSOaGTa
```
Comment (gomesalexandre) [inline .github/workflows/pr-cypress.yml:36]: Can't we make the API key programmatic in the consumption, using `REACT_APP_JUNOPAY_APP_ID`?

File: src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'fiatRampProviders', 'JunoPay'],
+})
+
+export async function getJunoPayAssets(): Promise<FiatRampAsset[]> {
+  try {
+    const junoPayData  = await axios.get(getConfig().REACT_APP_JUNOPAY_SUPPORTED_COINS)
+    
+    let junoPayToCurrencyList= junoPayData.data.data.settings.buy.to_currency
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts:16]: Lint will be very sad, also can we optional chain at least,`junoPayData?.data?.data` to ensure we try to access an actual response ?

File: src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    
+    let junoPayCurrencyList = junoPayToCurrencyList.map((item: any) => {
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts:18]: Would prefer pluralization 
```suggestion
    let junoPayCurrenciesList = junoPayToCurrencyList.map((item: any) => {
```

**Comment by 0xApotheosis** [comment]
Environment variables added to Fleek: `app` and `develop`.

File: src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      return item;
+    });
+    let allCurrencyList = junoPayData.data.data.settings.metadata 
+    
+    const junoPayAssets = allCurrencyList.filter((item: { short_name: string }) => junoPayCurrencyList.includes(item.short_name.toUpperCase()));
+    
+    const assets: FiatRampAsset[] = junoPayAssets.map((asset: { long_name: string; short_name: string; logo_url: string; }) => ({
+      assetId: adapters.junopayTickerToAssetId(asset.short_name),
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts:26]: Not something that needs to be done in the scope of this PR but rather an overlook in the fiat ramps abstraction: If we make this a regular function body without the shorthand return syntax, and do
`const assetId = adapters.junopayTickerToAssetId(asset.short_name)`

Then we can actually surface the symbol, name, and asset image URL from our own`Asset` type.

File: src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      err,
+      { fn: 'getJunoPayAssets' },
+      'Get supported coins (JunoPay) failed',
+    )
+    return []
+  }
+}
+
+export const createJunoPayUrl = (action: FiatRampAction, asset: string, address: string): string => {
+  const JUNOPAY_BASE_URL = new URL('https://juno.finance/partners/shapeshift')
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/juno-pay.ts:44]: This isn't a base URL, use `REACT_APP_JUNOPAY_BASE_URL` from env as an actual base URL, append the `/partners/shapeshift` URI to it, and let's call it `junoPayPartnerURL`

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
A few early comments WRT naming and env vars, but overall looks good! 

Also, please run `yarn lint:fix` as lint is very sad currently

File: react-app-rewired/headers/csps/defi/junopay.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import type { Csp } from '../../types'
+
+export const csp: Csp = {
+  'connect-src': [
+    // https://github.com/shapeshift/web/blob/d076ba9764180cd8e6f18f28b5fd23e92796d559/src/components/Modals/FiatRamps/fiatRampProviders/coinbase-pay.ts#L41
```
Comment (0xdef1cafe) [inline react-app-rewired/headers/csps/defi/junopay.ts:5]: wrong comment

File: react-app-rewired/headers/csps/defi/junopay.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    process.env.REACT_APP_JUNOPAY_BASE_URL!,
+  ],
+  'img-src': [
+    // https://github.com/shapeshift/web/blob/70111acd6236759675cf81ddcd196c31472989a6/src/components/Modals/FiatRamps/utils.ts#L13
```
Comment (0xdef1cafe) [inline react-app-rewired/headers/csps/defi/junopay.ts:9]: wrong comment

**Comment by 0xdef1cafe** [comment]
hey @rakeshnuo - thanks for the PR! this is a super good foundation, for expediency i hope you don't mind if we take this one over internally from here to get it across the line quickly

closing in favor of https://github.com/shapeshift/web/pull/2129

**Pull request closed without merging**